libs/src/utils/apiSigning.js

```diff
@@ -2,14 +2,16 @@ const assert = require('assert')
 
 /**
  * Recover the public wallet address given the response contains the signature and timestamp
- * @param {object} response discovery provider response
+ * @param {object} response entire service provider response (not axios)
  */
 function recoverWallet (web3, response) {
   let ownerWallet = null
-  try {
-    const dataForRecovery = { data: response.data, timestamp: response.timestamp }
-    const dataForRecoveryStr = JSON.stringify(_sortKeys(dataForRecovery))
 
+  const dataForRecovery = JSON.parse(JSON.stringify(response))
```

Comment:
this is one way to clone the object. another easier way is
```js
const dataForRecovery = {...response}
```

Reply:
i was thinking that we could make a copy using the spread operator, but it would be a shallow copy of the response. for our usage in removing the `signature` key, it doesn't really matter but i thought it would be safer to create a fully independent, deep copy